### PR TITLE
add matomo tracking

### PR DESCRIPTION
**Pull request by achen2401**
https://www.pivotaltracker.com/story/show/183594028

siteId from environment variable, `REACT__MATOMO_SITE_ID`
userId from parsed token response from session Storage:

Screenshot 2024-12-04 at 11 24 12 AM

[parsed from token response](https://jwt.io/#debugger-io?token=eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJfcW5PSE1xeDhidGpldW1CQUVvNjBaQVJoWThxd2Y0WU1ncU1GOEZreHNFIn0.eyJleHAiOjE3MzMzMzg2MjksImlhdCI6MTczMzMzNjgyOSwiYXV0aF90aW1lIjoxNzMzMzM2ODIzLCJqdGkiOiJmZDEyOTlhOS1lNjk1LTRmZTItODVmMy03Zjk1NWE2YWMyMzUiLCJpc3MiOiJodHRwczovL2tleWNsb2FrLmNvc3JpLWRlbW8uY2lyZy53YXNoaW5ndG9uLmVkdS9hdXRoL3JlYWxtcy9mRU1SIiwiYXVkIjoiY29zcmlfb3BlbmlkX2NsaWVudCIsInN1YiI6Ijc5NGQyNzgyLWU3ODMtNGNiYi1hZTcyLWIxNjVhNGMwNmU3ZiIsInR5cCI6IklEIiwiYXpwIjoiY29zcmlfb3BlbmlkX2NsaWVudCIsInNlc3Npb25fc3RhdGUiOiJlNTM5MDU5OS1jMGU5LTQzZGMtYjc2ZC1kZjU1NjkxODFmOTEiLCJhdF9oYXNoIjoiYXRycWZ4YkxJOGYxOG44eEJZSHJXUSIsImFjciI6IjAiLCJzaWQiOiJlNTM5MDU5OS1jMGU5LTQzZGMtYjc2ZC1kZjU1NjkxODFmOTEiLCJERUEiOiJ0ZXN0REVBdmFsdWUiLCJlbWFpbF92ZXJpZmllZCI6ZmFsc2UsInByZWZlcnJlZF91c2VybmFtZSI6InRlc3QifQ.D-6H-jNvVQKKbB1dfrLxS684j9045bRd_kxlzqwWnzc6s9EhwRmEJPGSfvGvC5V9HuxpDttqCYcezvLO5R0XE-5HLKYjvnWnwqIbdug66b4cLihd0z1DYD7cRgRVKaWFZKPiFOvivV8X72IIWzmbWNNGEFRD-KqVuAj-YG62KWRqMg-5hGm6_knTUUkx50mGirg790mSJSPEBch8PIin8Z5bAJTM2V7fDv-KLsWQBdzKwAN751sfL4npo9sORdYLvT8R96ENp2AI72uc_A5kBtJL1KImIh8UjyPvqUtynT3JLQFdYJTXNTzJj74OqkH1G2gtVQ9CGsjikR1TfoTOmA&publicKey=%7B%0A%20%20%22e%22%3A%20%22AQAB%22%2C%0A%20%20%22kty%22%3A%20%22RSA%22%2C%0A%20%20%22n%22%3A%20%2232V0Qb_oKMLTmuoKB13IBhmvN_JIBANnuIp7APeIys3veoL_uSGbBpoSRcjqtLs4g2wqzOnEv-ZiFdd_9dnPJ1VR1L_C-JmL7VUi6tlgJWkEignh54NV8qS1eqZkOCwDZE0rAWZHrf0xhlib5pbL39maz4OCwlgG1uHy5w2RpAGnxB5uF2DxorMEfGm1eHyPgQsMx07YhQQWtqwo65QsZtgcbc8gCOKOP_-l5ehvBZIOgoEdxUJ81_opvPS9F1Hhh_P31IcfAf2AaczdXORCrI_4ZgSurco87jpeA-x0P4aQZd9fG4CtsQc_UUs4AxkoTbw0nH7Edf-2EI-ypOHjow%22%0A%7D)

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -1,19 +1,39 @@
...
+        var d = document,
+          g = d.createElement("script"),
+          s = d.getElementsByTagName("script")[0];
+        g.type = "text/javascript";
+        g.async = true;
+        g.defer = true;
+        g.src = u + "matomo.js";
+        s.parentNode.insertBefore(g, s);
+      })();
+    </script>
```
Comment (mcjustin) [inline public/index.html:33]: @achen2401  looks like you copied the js that I listed for Sie-Ce for the project website, but instead you want the js from here: https://www.pivotaltracker.com/story/show/183594028/comments/233910403

**Review by mcjustin — CHANGES_REQUESTED** [review]

File: public/index.html (hunk 1 above)
Comment (achen2401) [reply public/index.html]: thank you @mcjustin. I fixed it.

File: public/index.html [hunk 2]
```diff
@@ -1,19 +1,39 @@
...
+    />
+    <meta name="theme-color" content="#000000" />
+    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico" />
     <title>Washington Opioid Clinical Assessment</title>
+    <script type="text/javascript">
+      var _paq = window._paq || [];
+      /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
+      _paq.push(["trackPageView"]);
+      _paq.push(["enableLinkTracking"]);
```
Comment (mcjustin) [inline public/index.html]: You dropped this: `  _paq.push(["setUserId", "REPLACE THIS WITH THE USER ID, IF KNOWN']);` 
Don't we know that in fEMR?

Comment (achen2401) [reply public/index.html]: @mcjustin I am not able to easily obtain that here.  I don't know where to get the UserID from (e.g. from the JWT Token?)

**Review by ivan-c — APPROVED** [review]
Looks great, thanks!

File: src/components/Landing/index.js [hunk 3]
```diff
@@ -6,6 +6,7 @@ import executeElm from "../../utils/executeELM";
 import * as landingUtils from "./utility";
 import { datishFormat } from "../../helpers/formatit";
 import {
+  addMamotoTracking,
```
Comment (mcjustin) [inline src/components/Landing/index.js:9]: @achen2401  it's Matomo, not Mamoto - can you please edit so it's easier to find :)

Comment (achen2401) [reply src/components/Landing/index.js]: thank you @mcjustin for noticing the typo! I fixed it now.

File: src/helpers/utility.js [hunk 4]
```diff
@@ -707,3 +708,45 @@ export function dedupArrObjects(arr, key) {
     return acc;
   }, []);
 }
+
+export function getMamotoTrackingSiteId() {
```
Comment (mcjustin) [inline src/helpers/utility.js:712]: @achen2401  another Mamoto gremlin here :) - and a call to it below.

Comment (achen2401) [reply src/helpers/utility.js]: thank you @mcjustin! This should be fixed too.

**Review by mcjustin — APPROVED** [review]

**Pull request merged**